cmd/createcluster.go

```diff
@@ -430,6 +432,17 @@ var validNetworks = map[string]bool{
 
 // validateClusterConfig returns an error if the cluster config is invalid.
 func validateClusterConfig(conf clusterConfig) error {
+	if conf.NumNodes < defaultNumNodes {
+		return errors.New("insufficient number of nodes (min = 4)")
+	}
+
+	// minReq is the minimum allowable threshold.
+	minThres := cluster.Threshold(conf.NumNodes)
+	//nolint:wrapcheck
+	if conf.Threshold < minThres {
+		return fmt.Errorf("threshold less than minimum (min required >= %d)", minThres)
+	}
```

Comment:
```
safeThreshold := cluster.Threshold(conf.NumNodes)
if conf.Threshold == 0 {
  conf.Threshold = safeThreshold
} else if conf.Threshold != safeThreshold {
  log.Warn(ctx, "Non standard `--threshold` flag provided, this will affect cluster safety", nil, z.Int("threshold", conf.Threshold), z.Int("safe_threshold", safeThreshold ))
}

```